x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -367,7 +370,9 @@ export class Authenticator {
   async authenticate(request: KibanaRequest) {
     assertRequest(request);
 
-    const existingSessionValue = await this.getSessionValue(request);
+    const existingSessionValueRaw = await this.getSessionValue(request);
+    const existingSessionValue =
+      existingSessionValueRaw instanceof SessionError ? null : existingSessionValueRaw;
```

Comment:
Here (and other places in this file), I hide the potentially returned error in the "raw" version of the variable, as the rest of the code still expects either a session or `null`. I'm not sure if this is the best approach though.